crypto/codec/tm.go

```diff
@@ -0,0 +1,57 @@
+package codec
+
+import (
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/encoding"
+	tmprotocrypto "github.com/tendermint/tendermint/proto/tendermint/crypto"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// FromTmPublicKey converts a TM's tmprotocrypto.PublicKey into our own PubKey.
+func FromTmPublicKey(protoPk tmprotocrypto.PublicKey) (cryptotypes.PubKey, error) {
+	switch protoPk := protoPk.Sum.(type) {
+	case *tmprotocrypto.PublicKey_Ed25519:
+		return &ed25519.PubKey{
+			Key: protoPk.Ed25519,
+		}, nil
+	default:
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "cannot convert %v from Tendermint public key", protoPk)
+	}
+}
+
+// ToTmPublicKey converts our own PubKey to TM's tmprotocrypto.PublicKey.
+func ToTmPublicKey(pk cryptotypes.PubKey) (tmprotocrypto.PublicKey, error) {
```

Comment:
This names are too similar: `ToTmPubKey` and `ToTmPublicKey`. Can we rename them please? One is a struct, the second one is an interface. So maybe adding some suffix?